### ISSUE TITLE
'powerpc-linux-gnu-gcc' can't be installed correctly

### ISSUE THREAD

**Issue by bet4it** [Jguer/yay#695]
#### Affected Version
yay v8.1115.r2.g8248c80 - libalpm v11.0.1

#### Issue
We can't use `yay` to install `powerpc-linux-gnu-gcc` correctly.
Some reasons may be that `yay` can't check for circular dependency correctly and `yay` doesn't resolve the right install order.

#### Steps to reproduce
1.`yay -S powerpc-linux-gnu-gcc`
2.Install the `powerpc-linux-gnu-binutils` and `powerpc-linux-gnu-linux-api-headers` dependency.
3.When `yay` is about to install `powerpc-linux-gnu-glibc`, we will get the following error:

> ==> Missing dependencies:
  -> powerpc-linux-gnu-gcc-stage2>=8.1.0-1
==> ERROR: Could not resolve all dependencies.
Error making: powerpc-linux-gnu-glibc

**Comment by Morganamilo** [comment]
The problem seems to come from `powerpc-linux-gnu-gcc` providing `powerpc-linux-gnu-gcc-stage2>=8.1.0-1`. This causes our dep resolver to be satisfied and not pull in `powerpc-linux-gnu-gcc-stage2`, Then our dep ordering doesn't have all the packages it needs to come up with a valid solution.

**Comment by bet4it** [comment]
Currently there are more similar packages in `AUR` like `arm-linux-gnueabihf-gcc` and `powerpc64-linux-gnu-gcc`.
Hope we can reopen this issue until it's really solved.

**Comment by jfalcou** [comment]
The issue is still impacting yay.

**Comment by bet4it** [comment]
Still valid

**Comment by pfrenssen** [comment]
Not stale.

**Comment by AchmadFathoni** [comment]
Is there any AUR helper that doesn't have this problem?

**Comment by AchmadFathoni** [comment]
> Is there any AUR helper that doesn't have this problem?

Nevermind, yay actually can install arm-linux-gnueabihf-gcc only with this installation sequence

arm-linux-gnueabihf-gcc-stage2
arm-linux-gnueabihf-gcc-stage1
arm-linux-gnueabihf-gcc

**Comment by bet4it** [comment]
> Is there any AUR helper that doesn't have this problem?

`aurman` with this [patch](https://github.com/polygamma/aurman/pull/262.patch)

**Comment by gkatev** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

don't close?

**Comment by Jguer** [comment]
Should be fixed in `yay-git`, if it's not (I think there might be an order dependent case) an extra step can be added to retry adding failed packages one more time to the graph.

Closing until more feedback

**Comment by gkatev** [comment]
Thanks @Jguer, I installed `yay-git` and tried to test this with `arm-linux-gnueabihf` (which is out of date). If I do `yay -Syu`, I get the following choice:

```
:: There are 2 providers available for arm-linux-gnueabihf-gcc-stage2>=12.1.0-1:
:: Repository AUR
    1) arm-linux-gnueabihf-gcc 2) arm-linux-gnueabihf-gcc-stage2 

Enter a number (default=1): 
==>
```
If I select the default one, the following mesasge gets repeatedly spammed to the console

```
aur dep warn:arm-linux-gnueabihf-gccarm-linux-gnueabihf-glibccircular dependencies not allowed
```

Now I'm not sure if I'm actually supposed to select `2`, though probably not because it is -gcc I want installed, not -gcc-stage2 (?). If I do choose 2, stage2 eventually gets installed succesfully. But if the fix worked this shouldn't be necessary?

**Comment by Talanor** [comment]
Hi. The issue persists to this day.
